src/services/create-project.service.js

```diff
@@ -123,6 +123,14 @@ export default (
             onComplete(packageJson);
           }
         );
+
+        if (projectType === 'create-react-app') {
+          // CRA 2.0 introduces functionality to immediately init a git repo
+          // Need to do a commit after creation to make sure the repo is clean
+          childProcess.exec('git add -A && git commit -m "Add to Guppy"', {
```

Comment:
Nit: I think we could also do it with `git commit -am "Add to Guppy"`.

I'm wondering what will happen if `Git` is not installed on the computer. I've seen on MacOS that it will prompt the user to install Git. e.g. typing `git` on a fresh OSx installation will trigger that prompt.
Not sure if the same will happen if CRA tries to init the Git repo.
So I think we don't need to add a check if Git is available here but I think we should do it before `npx create`.

For auto-committing I have an idea how to do it - I'll write something to it in the review comment.

Reply:
I feel like we should be more precise, with `git add package.json && git commit -m "Add Guppy data to package.json"`.

I know it shouldn't be possible to have any other dirty files since the project is brand-new, but it feels a bit safer in terms of future changes / edge-cases.

I also tweaked the message to be a bit more specific, although I don't feel too strongly about it.

Reply:
To @AWolf81's point about Git being installed, I was curious how CRA dealt with this possibility, and found [their code](https://github.com/facebook/create-react-app/blob/9e074bbf08a8760579e7345fe1391a76f9c41f96/packages/react-scripts/scripts/init.js#L43). Essentially they wrap it in a try/catch and do nothing if it fails.

I think we should do the same; since CRA won't create a Git repo if it fails, we shouldn't mind if trying to create a commit fails as well. So we should wrap it in a try/catch and swallow the error

Reply:
@joshwcomeau good to know how CRA is handling Git not available. 
Try/catch & swallowing the error is OK here. 
Should we move git commands into a `git.service` so we're prepared for a future git integration?  I think it would be also OK to add it here directly as we're having just this callsite.